data/src/main/java/org/apache/iceberg/data/GenericAppenderFactory.java

```diff
@@ -206,7 +252,7 @@ public EqualityDeleteWriter<Record> newEqDeleteWriter(
   @Override
   public PositionDeleteWriter<Record> newPosDeleteWriter(
       EncryptedOutputFile file, FileFormat format, StructLike partition) {
-    MetricsConfig metricsConfig = MetricsConfig.fromProperties(config);
+    MetricsConfig metricsConfig = applyMetricsConfig(() -> MetricsConfig.forPositionDelete(table));
```

Comment:
Could this work instead of an extra method?
```
MetricsConfig metricsConfig = table != null ? MetricsConfig.forPositionDelete(table): MetricsConfig.fromProperties(config);
```

For me, this is way easier to understand